### PR TITLE
Adds the ☼Floor Pill☼.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

See title.

## Why It's Good For The Game

Adds a degree of emergent gameplay and danger to maintenance.

## Changelog
:cl: Urumasi
add: Adds the ☼Floor Pill☼.
/:cl:

### CODE REVIEW

**Comment by 170141183460469231731687303715884105727** [comment]
Speedmerge.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 1]
```diff
@@ -295,3 +295,21 @@
...
+/obj/item/reagent_containers/pill/floor_pill
+	name = "☼Floor Pill☼"
+	desc = "This is a masterful floor pill. It is decorated with ashwalker bone and encircled with bands of iron and silver. The object menaces with spikes of wood. On the item is an image of an assistant eating a floor pill. The assistant is dying. On the item is an image of four medical doctors. The medical doctors are praying."
+	icon_state = "pill8"
+	list_reagents = list()
+	
+/obj/item/reagent_containers/pill/floor_pill/Initialize()
+	var/list/picked_reagents = list()
+	for(var/reagent in typesof(/datum/reagent/))
+		if(prob(rand(1,100)))
```
Comment (Urumasi) [inline code/modules/reagents/reagent_containers/pill.dm:308]: `prob(rand(1,100))` passes 50.5 % of the time

Comment (Iamgoofball) [reply code/modules/reagents/reagent_containers/pill.dm]: Resolved.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 2]
```diff
@@ -295,3 +295,22 @@
 	icon_state = "pill8"
 	list_reagents = list(/datum/reagent/iron = 30)
 	rename_with_volume = TRUE
+	
+/obj/item/reagent_containers/pill/floor_pill
+	name = "☼Floor Pill☼"
+	desc = "This is a masterful floor pill. It is decorated with ashwalker bone and encircled with bands of iron and silver. The object menaces with spikes of wood. On the item is an image of an assistant eating a floor pill. The assistant is dying. On the item is an image of four medical doctors. The medical doctors are praying."
+	icon_state = "pill8"
```
Comment (TemporalOroboros) [inline code/modules/reagents/reagent_containers/pill.dm:302]:
```suggestion
	icon_state = "pill8"
	reagent_flags = NO_REACT
```

I don't think you want this rolling RDX + Teslium and instantly breaching maint at roundstart.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 3]
```diff
@@ -295,3 +295,22 @@
...
+/obj/item/reagent_containers/pill/floor_pill/Initialize()
+	var/list/picked_reagents = list()
+	var/list/all_reagents = subtypesof(/datum/reagent/)
+	var/reagent_count = rand(1,all_reagents.len)
+	for(var/i in 1 to reagent_count)
+		picked_reagents += pick_n_take(all_reagents)
+	if(picked_reagents)
+		volume = (picked_reagents.len * 10)
+	for(var/picked_reagent in picked_reagents)
+		list_reagents[picked_reagent] = 10
```
Comment (TemporalOroboros) [inline code/modules/reagents/reagent_containers/pill.dm:314]:
```suggestion
	var/reagent_volume = 0
	if(picked_reagents)
		volume = min(picked_reagents.len * 10, 1000)
		reagent_volume = volume / picked_reagents.len
	
	for(var/picked_reagent in picked_reagents)
		list_reagents[picked_reagent] = reagent_volume
```

This way the reagents transferred aren't limited by the stomach capacity of the mob eating them unless they've eaten something else immediately before.

**Comment by Y0SH1M4S73R** [comment]
It doesn't seem like there are any ways for this item to spawn outside of admins spawning it in.

**Comment by Rohesie** [comment]
I assume this is a meme and will give it until it's 24 hours old to close it.

**Comment by optimumtact** [comment]
put some effort into randomising and generating the maint pill items

File: code/modules/reagents/reagent_containers/pill.dm (hunk 3 above)
Comment (Iamgoofball) [reply code/modules/reagents/reagent_containers/pill.dm:315]: Thanks!

**Comment by MrDoomBringer** [comment]
>I assume this is a meme and will give it until it's 24 hours old to close it.

It doesn't seem like a meme to me? It's silly/ridiculous, sure, but so are a lot of the more fun things in this game. I don't see a problem so long as it's not cropping up every round.

**Comment by optimumtact** [comment]
they didn't finish the PR so it looked like a meme

**Comment by Iamgoofball** [comment]
i forgot to push one of the changed files by accident

**Comment by Fikou** [comment]
because of how loot tables are set up, this will be more common to get than maint meth, maybe lower its weight a bit?

**Comment by Iamgoofball** [comment]
is the maint prob not 1 to 100?

**Comment by SmArtKar** [comment]
I suggest you put those super junky-dunky random pills into oddities and not rare loot because of how unpredictable they are

**Comment by ExcessiveUseOfCobblestone** [comment]
It looks like a weight table.

We have maint pills already (line 258) and they have some limitations on them that this pill will not have (or at least I thought I did, I never use them Ingame and mobile posting). At the very least they abide by can_synth.

Im just thinking why would I not crush that pill and never actually worry about bad chems but you’ve made it also hold upto 1000u so maybe it’ll get bricked trying on most beakers. 

We have can synth for a purpose, I’m not too keen on giving potentially Arcade pulse rifle tier loot in a drop that is guaranteed to run multiple times without anyone having to do anything roundstart.

**Comment by MrDoomBringer** [comment]
>We have maint pills already

This intentionally disregards that - by my understanding the joke is that it's a masterwork maint pill (hence the ☼ symbols) and as such is a truly unrestrained wild ride.

Sounds like you'd prefer if this was moved to the [oddity ](https://github.com/tgstation/tgstation/pull/48489)loot table? For a refresher, that would give it a 1:10,000 chance per spawner, or appearing about once in 40 rounds on Metastation.

File: code/modules/reagents/reagent_containers/pill.dm [hunk 4]
```diff
@@ -295,3 +295,25 @@
...
+	for(var/i in 1 to reagent_count)
+		picked_reagents += pick_n_take(all_reagents)
+	var/reagent_volume = 0
+	if(picked_reagents)
+		volume = min(picked_reagents.len * 10, 1000)
+		reagent_volume = volume / picked_reagents.len
+	
+	for(var/picked_reagent in picked_reagents)
+		list_reagents[picked_reagent] = reagent_volume
+	..()
```
Comment (Watermelon914) [inline code/modules/reagents/reagent_containers/pill.dm:319]:
```suggestion
	return ..()
```

**Comment by Fikou** [comment]
you should make them immune to being scanned/grinded

**Comment by Mothblocks** [comment]
> You are expected to have tested your pull requests if it is anything that would warrant testing. Text only changes, single number balance changes, and similar generally don't need testing, but anything else does. This means by extension web edits are disallowed for larger changes.

image

**Comment by Iamgoofball** [comment]
Excuse me? This code works.

**Comment by MrDoomBringer** [comment]
>This means by extension web edits are disallowed for larger changes.

Hello Iamgoofball. As a maintainer of this repository, you are in the position to act as a rolemodel for other contributors to follow. As suchc, we expect you to uphold the [/TG/Station Official Contributing Guidelines](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#pull-request-process) to a strict degree. The /TG/Station Official Contributing guidelines can be found at https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#pull-request-process. If you are unable to click on the above link, please copy and paste it into your browser's address bar.

>This means by extension web edits are disallowed for larger changes.

**Pull request closed without merging**